File: go/docmost/services/license_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// LicenseAPI implements License API.
type LicenseAPI struct {
	http *httpclient.HTTPClient
}

func NewLicenseAPI(http *httpclient.HTTPClient) *LicenseAPI {
	return &LicenseAPI{http: http}
}

func (a *LicenseAPI) GetLicenseInfo(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "license/info", nil)
}

func (a *LicenseAPI) ActivateLicense(ctx context.Context, request *models.ActivateLicenseDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "license/activate", request)
}

func (a *LicenseAPI) RemoveLicense(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "license/remove", nil)
}

File: python/docmost/services/license_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import ActivateLicenseDto


class LicenseApi:
    """License API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def get_license_info(self) -> ApiResponse[Any]:
        return self._http.post("license/info", body=None)

    def activate_license(self, request: ActivateLicenseDto) -> ApiResponse[Any]:
        return self._http.post("license/activate", body=request.model_dump(by_alias=True, exclude_none=True))

    def remove_license(self) -> ApiResponse[Any]:
        return self._http.post("license/remove", body=None)

